rosetta/scripts/scripts.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2021 Alvalor S.A.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package scripts
+
+import (
+	"strings"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+func WithParams(params Params) func(*Scripts) {
+	return func(s *Scripts) {
+		s.params = params
+	}
+}
+
+type Scripts struct {
+	params Params
+}
+
+func New(options ...func(*Scripts)) *Scripts {
+
+	s := Scripts{
+		params: TestNet(),
+	}
+
+	for _, option := range options {
+		option(&s)
+	}
+
+	return &s
+}
+
+func (s *Scripts) GetBalance(token flow.Address) []byte {
+	script := GetBalance
+	script = strings.ReplaceAll(script, PlaceholderFungible, s.params.FungibleToken.Hex())
+	script = strings.ReplaceAll(script, PlaceholderToken, token.Hex())
+	return []byte(script)
```

Comment:
Why not use [`text/template`](https://golang.org/pkg/text/template/) for this instead of having consts for each placeholder value? For now it is manageable but with more complex scripts it might get tricky to maintain/read.

Reply:
I don't have much experience with it, but it just seems overly complicated. The performance overhead here is really minimal, and I want to stay close to how they do it in the Flow Go SDK. You might want to create an issue to review this, though.

Reply:
What's nice with `text/template` is that it makes is very clear what is and what is not a _placeholder_ when reading the template itself, while here one might thing that `0xFUNGIBLE_TOKEN` is somehow a part of the file and is not going to be templated. But it's a detail IMO so it's fine like this for now I'd say